### ISSUE TITLE
Different classes on server and client when using css prop

### ISSUE THREAD

**Issue by th0th** [styled-components/styled-components#2538]
I get this warning when using styled-components 4.2.0 with next.js 8.

> Warning: Prop `className` did not match. Server: "sc-cJSrbW kQAtac" Client: "sc-htoDjs iZPNuM"

The only step needed to reproduce is to add something like this in a page on [the default next.js example](https://github.com/zeit/next.js/tree/canary/examples/with-styled-components):

```
<div
  css="margin-top: 20px;"
>
  Just some text
</div>
```

or 

```
<div
  css={css`margin-top: 20px;`}
>
  Just some text
</div>
```

Here is a codesandbox:
https://codesandbox.io/embed/jlwvwyy0ow
https://jlwvwyy0ow.sse.codesandbox.io/

![Annotation 2019-05-10 045959](https://user-images.githubusercontent.com/698079/57497472-7b657c00-72e0-11e9-84c5-e7e5fa0d351c.png)

**Comment by probablyup** [comment]
There's something weird going on with nextjs in dev mode (I see this in my projects as well), but the issue does not go through to production in my experience.

**Comment by th0th** [comment]
Thanks for the reply @probablyup. So do you just ignore those warnings in development mode? The number of warnings goes up as I keep adding more `css` props and it is annoying :)

**Comment by probablyup** [comment]
Yeah I’ve just been ignoring them. We are removing that warning anyway so
don’t worry about it 👍

**Comment by timneutkens** [comment]
We don't really do anything weird here, just webpack HMR and re-rendering. Could you file an issue on Next.js so that I can add labels for people to look into it.

**Comment by th0th** [comment]
@timneutkens Here it is: https://github.com/zeit/next.js/issues/7322

**Comment by probablyup** [comment]
@th0th this should be fixed with the latest babel plugin patch version, thanks for reporting! https://github.com/styled-components/babel-plugin-styled-components/releases/tag/v1.10.1
